src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,132 @@
+    private static final int BELOW_MIN_AGE = 17;
+    private static final int MIN_AGE = 18;
+    private static final String VALID_PASSWORD = "TheStrongestPassword";
+    private static final String VALID_LOGIN = "NewUser";
+    private static final int VALID_AGE = 30;
+    private static final String EMPTY_STRING = "";
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void invalidPassword_NotOk() {
+        user = new User(VALID_LOGIN, SHORT_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void invalidLogin_NotOk() {
+        user = new User(SHORT_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void theAgeIsLowerThanEighteen_NotOk() {
+        user = new User(VALID_LOGIN, VALID_PASSWORD, BELOW_MIN_AGE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void theAgeIsEighteen_Ok() {
+        user = new User(VALID_LOGIN, VALID_PASSWORD, MIN_AGE);
+        assertEquals(18, user.getAge());
+        assertDoesNotThrow(() -> registrationService.register(user));
+    }
+
+    @Test
+    void validUser_Ok() {
+        user = new User(VALID_LOGIN,VALID_PASSWORD, VALID_AGE);
+        assertDoesNotThrow(() -> registrationService.register(user));
+    }
+
+    @Test
+    void passwordLengthEqualsSix_Ok() {
+        user = new User(VALID_LOGIN,MIN_LENGTH_PASSWORD,VALID_AGE);
+        assertEquals(6, user.getPassword().length());
+        assertDoesNotThrow(() -> registrationService.register(user));
+    }
+
+    @Test
+    void passwordLengthEqualsZero_NotOk() {
+        user = new User(VALID_LOGIN, EMPTY_STRING,VALID_AGE);
+        assertEquals(0, user.getPassword().length());
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void loginLengthEqualsSix_Ok() {
+        user = new User(MIN_LENGTH_LOGIN,VALID_PASSWORD,VALID_AGE);
+        assertEquals(6, user.getLogin().length());
+        assertDoesNotThrow(() -> registrationService.register(user));
+    }
+
+    @Test
+    void loginLengthEqualsZero_NotOk() {
+        user = new User(EMPTY_STRING,VALID_PASSWORD, VALID_AGE);
+        assertEquals(0, user.getLogin().length());
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void userNotUnique_NotOk() {
+        user = new User(VALID_LOGIN,VALID_PASSWORD,VALID_AGE);
+        Storage.people.add(user);
+        assertThrows(RegistrationException.class, ()
+                -> registrationService.register(new User(VALID_LOGIN,VALID_PASSWORD,VALID_AGE)));
```

Comment:
```suggestion
                -> registrationService.register(new User(VALID_LOGIN, VALID_PASSWORD, VALID_AGE)));
```
use auto formatted to fix formatting mistakes before committing (Ctrl Alt L for windows machines)

Reply:
you could use the same object user to register the second time